src/engine/console.cpp

```diff
@@ -17,8 +18,29 @@ void complete(char* s, size_t s_size, const char* cmdprefix);
 
 void conline(int type, const char *sf, int n)
 {
-    char *buf = conlines.length() >= MAXCONLINES ? conlines.remove().cref : newstring("", BIGSTRLEN-1);
-    cline &cl = conlines.add();
+    // prevent that chat and console from 'scrolling' if user not at the bottom
+    if ((hud::chatpos != 0) && (hud::chatpos < hud::max_chatpos()))
+    {
+        hud::chatpos++;
+    }
+    if ((hud::consolepos != 0) && (hud::consolepos < hud::max_consolepos()))
+    {
+        hud::consolepos++;
+    }
+
+    bool add_to_chatlines = type >= CON_CHAT;
+
```

Comment:
Why new line here?

Reply:
that seems to be a mistake, happens sometimes :D, will fix